### DISCUSSION TITLE
Buypass provider support

### DISCUSSION THREAD

**Discussion by vkosuri** [go-acme/lego#2047] [Q&A] [answered]
I am trying to find the buypass https://api.buypass.com/acme/directory ACMEv2 provider support using lego, unfortunately, I did not find it, is there any plans or it's already supported?

**Comment by ldez** [comment, marked as answer]
Hello,

you can use any CA by using the option `--server`.
https://go-acme.github.io/lego/usage/cli/options/

**Reply by vkosuri** [reply]
could you give me an example to do this in Go?

**Reply by ldez** [reply]
The field `config.CADirURL`: https://go-acme.github.io/lego/usage/library/

**Reply by vkosuri** [reply]
Thanks I figured it. I have been doing this for ZeroSSL, BuyPass, LE, does this apply to all other supported CAs Sectio, SSL, etc listed here https://github.com/https-dev/docs/blob/master/list-of-acme-servers.md

```GO
switch certOptions.CAProvider {
	case "letsencrypt":
		caDirURL = "https://acme-v02.api.letsencrypt.org/directory"
	case "buypass":
		caDirURL = "https://api.buypass.com/acme/directory"
	case "zerossl":
		caDirURL = "https://acme.zerossl.com/v2/DV90"
	default:
		log.Fatal("Unknown CA provider specified in the config.json file")
	}
```
